### PR TITLE
Oomph setup

### PR DESCRIPTION
**Pull request by ruspl-afed**
Initial version of setup script
Target definition uses existing file but not yet activated for some reason

### CODE REVIEW

File: build/org.eclipse.birt.releng/BIRT.setup [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  </setupTask>
+  <setupTask
+      xsi:type="projects:ProjectsImportTask">
+    <sourceLocator
+        rootFolder="${git.clone.birt.location}"
+        locateNestedProjects="true"/>
+  </setupTask>
+  <setupTask
+      xsi:type="pde:TargetPlatformTask"
+      name="org.eclipse.birt.target"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.releng/BIRT.setup:155]: This does not work for some reason

**Comment by merks** [comment]
You have to use the value of the name attribute, not the file name:
```
<target includeMode="feature" name="BIRT Photon target platform">
```
It's looked up like this:

https://git.eclipse.org/c/oomph/org.eclipse.oomph.git/tree/plugins/org.eclipse.oomph.util.pde/src/org/eclipse/oomph/util/pde/TargetPlatformUtil.java#n246

**Comment by ruspl-afed** [comment]
> You have to use the value of the name attribute, not the file name:
> 

Thank you Ed @merks , it works after switching to attribute name

**Comment by ruspl-afed** [comment]
@wimjongman I have "just" 113 compilation errors after applying this script that hopefully not related with script itself, please have a look.

**Comment by wimjongman** [comment]
Are the compilation errors in a test bundle?

**Comment by ruspl-afed** [comment]
@wimjongman with #678 applied I have 89 Java errors all in `org.eclipse.birt.report.viewer.tests/test/org/eclipse/birt/report/viewer/mock`

**Review by wimjongman — APPROVED** [review]

**Comment by wimjongman** [comment]
Thanks! Please see the windowbuilder README.MD file. Can you add an entry like that?

https://github.com/eclipse/windowbuilder/blob/master/README.md#2-create-an-eclipse-development-environment

**Comment by wimjongman** [comment]
> 
> 
> @wimjongman with #678 applied I have 89 Java errors all in `org.eclipse.birt.report.viewer.tests/test/org/eclipse/birt/report/viewer/mock`

Should be fixed with https://github.com/eclipse/birt/pull/680

**Comment by ruspl-afed** [comment]
> Thanks! Please see the windowbuilder README.MD file. Can you add an entry like that?
> 
> https://github.com/eclipse/windowbuilder/blob/master/README.md#2-create-an-eclipse-development-environment

@wimjongman you can have a look here: https://github.com/eclipse/birt/tree/oomph_setup

**Comment by wimjongman** [comment]
I get an error message: 
firefox_jTTfqTt3l3

**Comment by ruspl-afed** [comment]
I'm afraid we can test it only after merging, since the URL has reference to master. Or do you want me to remove readme change and create dedicated PR @wimjongman ?

**Comment by wimjongman** [comment]
Here is the URL with reference to oomph_setup branch

https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse/birt/oomph_setup/build/org.eclipse.birt.releng/BIRT.setup&show=true

**Comment by ruspl-afed** [comment]
> Here is the URL with reference to oomph_setup branch
> 
> https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse/birt/oomph_setup/build/org.eclipse.birt.releng/BIRT.setup&show=true

Ooops. You are right. I think you need to click "next" on installer to go to "Project" page and then apply the URL. On the page from screenshot you selects what product should "host" the install.

**Comment by wimjongman** [comment]
Yes, on the second page I have to create a project with this setup URL. For windowbuilder I don't need to do that...

The second setup task asks me to install wtp. Should we not begin with that if it is required?

**Comment by wimjongman** [comment]
Can we also apply the project set file from the root of the directory?

**Comment by ruspl-afed** [comment]
> For windowbuilder I don't need to do that

I'll take a look what is the magic there

> The second setup task asks me to install wtp

No idea why, never seen this before. But I was testing it via Eclipse Installer 

image

> Can we also apply the project set file from the root of the directory?

It somehow managed to escape my pattern "In same repository as org.eclipse.birt"

**Comment by ruspl-afed** [comment]
> > For windowbuilder I don't need to do that
> 
> I'll take a look what is the magic there
> 

I looked at the windowbuilder script. `WindowBuilderConfiguration.setup` is a kind of wrapping "Product" script around another "Project" script `WindowBuilder.setup`

Currently we have only "Project" script for BIRT and need to add a "Product" one to make this button fully usable

**Comment by wimjongman** [comment]
Ok, #680 is merged and a good bunch of tests is running again. I'm moving on to fixing the next set of tests. With #680 merged, we should see no more errors after oomph is done.

**Comment by wimjongman** [comment]
Alexander, can you rebase this one?

**Comment by wimjongman** [comment]
Good to go, Alexander?

**Comment by ruspl-afed** [comment]
Wim, I would merge it.
And then I can try to create another one for "Product".

**Comment by wimjongman** [comment]
Thanks!!! 

By the way, "Signed-off-by:" is no longer required.

**Pull request merged**